native/cocos/2d/renderer/Batcher2d.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+#include <2d/renderer/UIMeshBuffer.h>
+#include <cocos/2d/renderer/RenderDrawInfo.h>
+#include <cocos/2d/renderer/UIMeshBuffer.h>
+#include <cocos/base/TypeDef.h>
+#include <cocos/core/assets/Material.h>
+#include <cocos/renderer/gfx-base/GFXTexture.h>
+#include <cocos/renderer/gfx-base/states/GFXSampler.h>
+#include <cocos/scene/DrawBatch2D.h>
+#include <unordered_map>
+#include <vector>
+#include "base/Ptr.h"
+
+namespace cc {
+class Root;
+class Batcher2d {
+public:
+    Batcher2d();
+    explicit Batcher2d(Root* root);
+    ~Batcher2d();
+
+    void syncMeshBuffersToNative(std::vector<UIMeshBuffer*>&& buffers, uint32_t length);
+
+    bool initialize();
+    void update();
+    void uploadBuffers();
+    void reset();
+    void updateVertDirtyRenderer();
+
+    void addRootNode(Node* node);
+
+public:
```

Comment:
Just one `public` is enough.

Reply:
keep an extra private ,one for attributes, another for inline functions